contracts/tokenomics/maker/src/contract.rs

```diff
@@ -452,19 +479,63 @@ fn distribute_astro(deps: DepsMut, env: Env, info: MessageInfo) -> Result<Respon
 /// # Params
 /// * **deps** is the object of type [`DepsMut`].
 ///
-/// * **cfg** is the object of type [`Config`].
+/// * **env** is the object of type [`Env`].
 ///
-/// * **amount** is the object of type [`Uint128`].
-fn distribute(deps: Deps, cfg: &Config, amount: Uint128) -> Result<Vec<SubMsg>, ContractError> {
+/// * **cfg** is the object of type [`Config`].
+fn distribute(deps: DepsMut, env: Env, cfg: &mut Config) -> Result<Vec<SubMsg>, ContractError> {
     let mut result = vec![];
 
+    let astro = token_asset_info(cfg.astro_token_contract.clone());
+
+    let mut amount = astro.query_pool(&deps.querier, env.contract.address.clone())?;
+    if amount.is_zero() {
+        return Ok(result);
+    }
+
+    if !cfg.rewards_enabled {
+        cfg.pre_upgrade_astro_amount = amount;
+        CONFIG.save(deps.storage, cfg)?;
+        return Ok(result);
+    } else if !cfg.pre_upgrade_astro_amount.is_zero() {
```

Comment:
maybe let's use remainder_reward.is_some and !remainder_reward.is_zero instead in this if, then we won't have to set cfg.pre_upgrade_astro_amount to zero to disable this branch and will keep the pre_upgrade_astro_amount for history